bricks/stm32/main.c

```diff
@@ -328,6 +359,12 @@ static void run_user_program(uint32_t len, uint8_t *buf, uint32_t free_len) {
 }
 
 static void stm32_main(void) {
+
+    #if PYBRICKS_HUB_PRIMEHUB
+    mp_hal_delay_ms(500);
```

Comment:
what is this delay for?

Reply:
For some reason Bluetooth wouldn't come up without it.

This initial read/write operation takes fairly long, so maybe there's still a bit too much blocking going on while everything else is initializing.